### ISSUE TITLE
Development Branch

### ISSUE THREAD

**Issue by ziggysauce** [chingu-coders/Voyage2-Turtles-02#106]
## Proposal
To create a separate `development` branch for new development and test

## Current Behavior
Currently all changes are pushed to `master` branch

## Issues
Now that our code is live on both the Google Chrome Store and Firefox Add-ons, we don't want to break our `master` branch code. The safer method is to create a separate `development` branch with extensive testing before merging to our `master` branch code. @TylerDelRosario I know you have experience with this from your previous work, and @jmbothe I'm sure you're familiar with it as well, I just haven't incorporated it into any of my repos yet and think it will be a good idea and good practice for future projects.

## Possible solutions
* Create new `development` branch
* On your own local device, create your own branch (bug fix, new feature, etc.)
* PR your changes to the `development` branch so that other members can test the new code
* After all code has been tested and ready for production, merge changes from `development` branch into `master` branch

**Comment by jmbothe** [comment]
Good idea! I just went ahead and created development branch in this repo. Everybody be on your toes: make sure you are making PRs to the correct branch!

**Comment by TylerDelRosario** [comment]
Good idea! A branch based workflow is alot more convenient.

**Comment by jmbothe** [comment]
Tips for a smooth workflow:

1. Before you start working, make sure you are on your local clone's `development` branch: `git checkout development`
2. Make sure that branch is tracking upstream's `development` branch: `git branch -u upstream/development`
3. Make sure your branch is even with upstream's branch: `git pull upstream development`
4. When you push to `origin` make sure you push to `development`, not `master`: `git push origin development`
5. When you make a PR, make sure you are comparing your fork's `development` with upstream's `development`
